packages/x-data-grid/src/hooks/features/rows/useGridRows.ts

```diff
@@ -239,6 +215,37 @@ export const useGridRows = (
     [props.signature, props.getRowId, throttledRowsChange, apiRef],
   );
 
+  const updateServerRows = React.useCallback<GridRowProApi['updateServerRows']>(
```

Comment:
Perhaps adding optional `groupKeys` to `updateRows` would be a better option here?

Reply:
[Here](https://github.com/mui/mui-x/pull/12317#discussion_r1633145117) are my arguments for making it a separate method. Let me know if it makes sense.

Reply:
I imagine it can make a few customizations possible in the userland.
For example, a CRUD implementation uses the Read part of the data source but manually updates rows using this API method.
We can just make it private though until we have a demand.

Reply:
> I'm not sure about the name though

Me neither. `updateServerRows` seemed to me most reasonable. `updateDataSourceRows` did not match with my mental model as we are not updating rows belonging to the `dataSource` itself. 😄 
Let me know in case of disagreements.